src/container/srv_container.c

```diff
@@ -703,10 +737,35 @@ cont_destroy(struct rdb_tx *tx, struct ds_pool_hdl *pool_hdl,
 		D_GOTO(out, rc);
 	}
 
+	/* Fetch the container props to check access for delete */
+	rc = cont_prop_read(tx, cont,
+			    DAOS_CO_QUERY_PROP_ACL |
+			    DAOS_CO_QUERY_PROP_OWNER |
+			    DAOS_CO_QUERY_PROP_OWNER_GROUP, &prop);
+	if (rc != 0)
+		D_GOTO(out, rc);
+	D_ASSERT(prop != NULL);
+
+	get_cont_prop_access_info(prop, &owner, &acl);
+
+	/*
+	 * Two groups of users can delete a container:
+	 * - Users who can delete any container in the pool
+	 * - Users who have been given access to delete the specific container
+	 */
+	if (!ds_sec_pool_can_delete_cont(pool_hdl->sph_sec_capas) &&
```

Comment:
should above "&&" be "||" actually?

Reply:
"&&" is correct in this case - if the caller has neither permission, we should reject the request. If they have one or the other, or both, they are allowed to delete the container.